src/expressions/Case.test.ts

```diff
@@ -63,9 +63,19 @@ describe("Case", () => {
         `);
 
         expect(queryResult.params).toMatchInlineSnapshot(`
-{
-  "param0": "Hello",
-}
-`);
+            {
+              "param0": "Hello",
+            }
+        `);
+    });
+
+    test("case with missing then fails", () => {
+        const testParam = new Cypher.Param("Hello");
+
+        const caseClause = new Cypher.Case().when(Cypher.eq(new Cypher.Literal("Hello"), testParam));
+
+        expect(() => {
+            new TestClause(caseClause).build();
+        }).toThrowError();
```

Comment:
I completely forgot that we have this error at build time! 
Not related to this PR, but: Perhaps we could raise an error as well in the Partial patterns in case of incomplete patterns?

Reply:
That's a good idea, but not too easy at the moment, due to how the partial patterns are built :sweat: 

We could do the check on the clauses the use it though